.github/workflows/render-all.yml

```diff
@@ -0,0 +1,165 @@
+
+# Candace Savonen Apr 2021
+
+name: Render all output courses
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main, staging ]
+    paths:
+      - '**.Rmd'
+      - assets/*
+
+jobs:
+
+  yaml-check:
+    name: Load user automation choices
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+        # Use the yaml-env-action action.
+      - name: Load environment from YAML
+        uses: doughepi/yaml-env-action@v1.0.0
+        with:
+            files: config_automation.yml # Pass a space-separated list of configuration files. Rightmost files take precedence.
+    outputs:
+      toggle_bookdown: "${{ env.RENDER_BOOKDOWN }}"
+      toggle_coursera: "${{ env.RENDER_COURSERA }}"
+      toggle_leanpub: "${{ env.RENDER_LEANPUB }}"
+
+  render-bookdown:
+    needs: yaml-check
+    runs-on: ubuntu-latest
+    container:
+      image: jhudsl/course_template
+    if: ${{needs.yaml-check.outputs.toggle_bookdown == 'yes'}}
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GH_PAT }}
+
+      - name: Login as jhudsl-robot
+        run: |
+          git config --local user.email "itcrtrainingnetwork@gmail.com"
+          git config --local user.name "jhudsl-robot"
+
+      # We want a fresh run of the renders each time
+      - name: Delete old docs/*
+        run: rm -r docs/*
+
+      # Run bookdown rendering
+      - name: Run bookdown render
+        id: bookdown
+        run: |
+          Rscript -e "bookdown::render_book('index.Rmd', output_format = 'all');
+          file.copy(from = 'assets', to = 'docs/assets', overwrite = TRUE)"
+
+      # This checks on the steps before it and makes sure that they completed.
+      # If the renders didn't complete we don't want to commit the file changes
+      - name: Check on render step
+        if: steps.bookdown.outcome != 'success'
+        run: |
+          echo Bookdown status ${{steps.bookdown.outcome}}
+          exit 1
+
+      # Commit the rendered bookdown files
+      - name: Commit rendered bookdown files
+        run: |
+          git config --local user.email "actions@github.com"
+          git config --local user.name "GitHub Actions"
+          git add --force docs/*
+          git commit -m 'Render bookdown' || echo "No changes to commit"
+          git push origin main || echo "No changes to push"
```

Comment:
I'm assuming this is just copied over from the old yamls? Did you make any big changes?

Reply:
Just copied over. GitHub can't track that apparently.